src/daemon/metric.c

```diff
@@ -511,7 +559,21 @@ metric_family_t *metric_family_clone(metric_family_t const *fam) {
     return NULL;
   }
 
-  status = metric_list_clone(&ret->metric, fam->metric, ret);
+  return ret;
+}
+
+metric_family_t *metric_family_clone(metric_family_t const *fam) {
+  if (fam == NULL) {
+    errno = EINVAL;
+    return NULL;
+  }
+
+  metric_family_t *ret = metric_family_clone_shallow(fam);
```

Comment:
`metric_family_clone_shallow()` also do this check, so it could be dropped from here:
```suggestion
  metric_family_t *ret = metric_family_clone_shallow(fam);
```
But compiler should optimize it away, so you can just resolve this, if you'd rather keep it.

Reply:
A good rule of thumb is that less code contains fewer bugs. Committed.